### ISSUE TITLE
reduce monorepo footprint

### ISSUE THREAD

**Issue by jondot** [loco-rs/loco#881]
In order to make build folder take less space, and to ease up on rust-analyzer, we'd like to

* [ ] Move `loco-cli` to its own independent repo (this was part of the plan until it stabilizes)
* [x] Move or remove `macros`, this was an experimental project that did not take off
* [x] Move `loco-extras` to its own repo, make sure that we still have interconnectedness between `loco` and `loco-extras` in terms of breakages
* [ ] refactor + simplify `xtask` bump version process https://github.com/loco-rs/loco/pull/893
* [ ] add a 'publish' step for subcrates, then main crate via `xtask`

**Comment by joshka** [comment]
I really like that you have a mono-repo - it's a plus when you can just make changes and have them coherent with the rest of the repo. This is fairly small project all things considered - larger by at least two orders of magnitude git repos are still good for monorepos IMHO.

What's the problem with r-a that you're hitting?

**Comment by jondot** [comment]
@joshka you're right, better to keep a monorepo. right now i'm hitting a couple of limits:
* `target/` folder size can reach 10-40GB on occasions
* rust-analyzer becomes sluggish as we're growing the list of members in the workspace

nevertheless, I've just added a subcrate: `loco-gen` because I had no other choice. For this instance, `loco-gen` will be the home of all things generators, including -- and more importantly -- generator testing, and generator testing was so heavy and filled with side-effects I had to isolate it in the form of a separate crate.

so looks like we're going to grow the amount of crates in the workspace any way, but then we'll remove some unneeded ones.
